### ISSUE TITLE
[Enhancement] Item Upgrade using nbt tags

### ISSUE THREAD

**Issue by DevDyna** [P3pp3rF1y/SophisticatedCore#175]
[like Thermal compounds on Thermal machines] 
Add custom nbt tags to allow the creation of custom upgrades based on `src/main/java/net/p3pp3rf1y/sophisticatedcore/upgrades` without need to code to add custom

Example create stack tiers with custom sizes only adding on any item `{Upgrade:{type:"stack",proprierties:{tier:24}}`

**Comment by P3pp3rF1y** [comment]
I can only see this applicable to stack upgrades really because they seem to be the only ones who don't expose one of their properties like this. I guess I could potentially do something like this for them, just don't see a huge reason for that unless you have ideas for how this would be useful / what is being blocked by current state.

**Comment by DevDyna** [comment]
I agree that can break too many stuff without any good reason and it was why i only specified on example the stack upgrades 
(because i didn't know if can be too huge or not)
